05.wc/wc.rb

```diff
@@ -0,0 +1,71 @@
+# ! /usr/bin/env ruby
+# frozen_string_literal: true
+
+require 'optparse'
+
+def main
+  argv_options = parse_and_remove_options(ARGV)
+  output_text = []
+  ARGF.each(nil) do |input_text|
+    wc_info = []
+    file_name = ARGF.filename
+    wc_info << [:filename, file_name]
+    wc_info << [:line, input_text.lines.count] if argv_options[:l]
+    wc_info << [:word, input_text.split(/\s+/).size] if argv_options[:w]
+    wc_info << [:bytesize, input_text.size] if argv_options[:c]
+    output_text << wc_info.to_h
+  end
+  output_text << file_total_info(output_text) if output_text.size > MULTIPLE_DISPLAY
+  max_width = get_max_width(output_text)
+  print_output(output_text, max_width)
+end
+
+DISPLAY_ADJUST_NUMBER = 1
+MULTIPLE_DISPLAY = 2
+TOTAL = '合計'
+
+def file_total_info(output_text)
+  file_total_info = output_text.flat_map(&:to_a).group_by(&:first).reject { |k, _v| k == :filename }
+  file_total_info.transform_values { |value| value.map(&:last).sum }.merge(filename: TOTAL)
+end
+
+def print_output(output_text, max_width)
+  output_text.each do |output_line|
+    puts format_output_line(output_line, max_width)
+  end
+end
+
+def format_output_line(output_line, max_width)
+  line = output_line[:line]&.to_s&.rjust(max_width[:line_display_width])
+  word = output_line[:word]&.to_s&.rjust(max_width[:word_display_width])
+  bytesize = output_line[:bytesize]&.to_s&.rjust(max_width[:bytesize_display_width])
+  filename = output_line[:filename] == '-' ? '' : output_line[:filename]
+
+  "#{line}#{word}#{bytesize} #{filename}"
+end
+
+def get_max_width(output_text)
+  max_lines_width = output_text.map { |entry| entry[:line].to_s.length + DISPLAY_ADJUST_NUMBER }.max || 0
+  max_words_width = output_text.map { |entry| entry[:word].to_s.length }.max + DISPLAY_ADJUST_NUMBER || 0
```

Comment:
```suggestion
  max_words_width = output_text.map { |entry| entry[:word].to_s.length + DISPLAY_ADJUST_NUMBER }.max || 0
```

ここだけこうしないのはなぜ？と思いました